src/iexamine.cpp

```diff
@@ -1832,88 +1894,219 @@ void iexamine::aggie_plant(player &p, const tripoint &examp)
+        pmenu.addentry( 1, true,  MENU_AUTOASSIGN, _( "add water          %s" ), water_info );
+        pmenu.addentry( 2, true,  MENU_AUTOASSIGN, _( "add fertilizer     %s" ), fertilizer_info );
+        pmenu.addentry( 3, true,  MENU_AUTOASSIGN, _( "remove weeds       %s" ), weed_info );
+        if( p.get_skill_level( skill_survival ) >= 2 ) {
+            pmenu.addentry( 0, false, MENU_AUTOASSIGN, health_info );
+            pmenu.addentry( 0, false, MENU_AUTOASSIGN, grow_info );
+            if( !growing_was_started ) {
+                pmenu.addentry( 0, false, MENU_AUTOASSIGN, _( "It is too cold to grow." ) );
             }
-            // Choose fertilizer from list
-            int f_index = 0;
-            if (f_types.size() > 1) {
-                f_names.push_back(_("Cancel"));
-                f_index = menu_vec(false, _("Use which fertilizer?"), f_names) - 1;
-                if (f_index == (int)f_names.size() - 1) {
-                    f_index = -1;
-                }
+            if( is_mature ) {
+                pmenu.addentry( 0, false, MENU_AUTOASSIGN, _( "The plant is mature and will yield fruits." ) );
+            }
+        }
+        pmenu.query();
+
+        int action_index = pmenu.ret;
+
+        // Watering
+        if( action_index == 1 ) {
+            const inventory &crafting_inv = p.crafting_inventory();
+            int water_charges = crafting_inv.charges_of( "water" );
+            if( water_charges == 0 ) {
+                add_msg( _( "You don't have water." ) );
+                return;
+            } else if( water > 400 ) {
+                add_msg( _( "There is no need to water this plant." ) );
+                return;
             } else {
-                    f_index = 0;
+                // Up to 50L of water can be used per plant
+                int water_use = std::min( water_charges * 5, ( 500 - water ) );
+                water += water_use;
+                seed.set_var( "water", water );
+                float water_used = water_use / 5.0f;
+                std::vector<item_comp> comps;
+                comps.push_back( item_comp( "water", std::max( static_cast<int>( water_used ), 1 ) ) );
+                p.consume_items( comps );
+                add_msg( _( "You watered the plant." ) );
+                p.moves -= 500;
+                return;
             }
-            if (f_index < 0) {
+        }
+
+        // Use fertilizer
+        if( action_index == 2 ) {
+            if( fertilizer > 300 ) {
+                add_msg( _( "There is no need to fertilize this plant." ) );
+                return;
+            } else {
+                std::vector<const item *> f_inv = p.all_items_with_flag( "FERTILIZER" );
+                if( f_inv.empty() ) {
+                    add_msg( m_info, _( "You have no fertilizer for the %s." ), pname.c_str() );
+                    return;
+                }
+
+                if( query_yn( _( "Fertilize the %s" ), pname.c_str() ) ) {
```

Comment:
There is confirmation query for fertilization, but no for watering.

Reply:
This confirmation query was removed, because player must select fertilizing action. It will not start by simple examine the plant (as now).